scripts/run_gen.py

```diff
@@ -1633,25 +1627,20 @@ def gen_and_test(num, makefile, lock, end_time, task_queue, stat, targets, blame
             continue
 
         # Run all required opt-sets.
-        out_res = set()
-        prev_out_res_len = 1  # We can't check first result
         for t in gen_test_makefile.CompilerTarget.all_targets:
-            # Skip the target we are not supposed to run.
-            if t.specs.name not in targets:
-                continue
-            target_elapsed_time = 0.0
-
-            test_run = TestRun(test=test, stat=stat, target=t, proc_num=num,
-                               parse_stats= True if (t.name in stat_targets) else False)
-            if not test_run.build():
-                test.add_fail_run(test_run)
-                continue
-
-            if not test_run.run():
-                test.add_fail_run(test_run)
-                continue
-
-            test.add_success_run(test_run)
+            # Run only specified targets
+            if t.specs.name in targets:
```

Comment:
I think it makes more sense to skip targets that are not specified upfront.
In this case, we don't have an extra offset, and it makes it more straightforward that there is no other code to handle them.

Reply:
Done